planner/core/plan_to_pb.go

```diff
@@ -285,12 +285,24 @@ func (p *PhysicalBroadCastJoin) ToPB(ctx sessionctx.Context, storeType kv.StoreT
 	case AntiLeftOuterSemiJoin:
 		pbJoinType = tipb.JoinType_TypeAntiLeftOuterSemiJoin
 	}
+	probeFiledTypes := make([]*tipb.FieldType, 0, len(p.EqualConditions))
+	buildFiledTypes := make([]*tipb.FieldType, 0, len(p.EqualConditions))
+	for _, equalCondition := range p.EqualConditions {
+		retType := equalCondition.RetType.Clone()
+		chs, coll := equalCondition.CharsetAndCollation(ctx)
+		retType.Charset = chs
+		retType.Collate = coll
+		probeFiledTypes = append(probeFiledTypes, expression.ToPBFieldType(retType))
+		buildFiledTypes = append(buildFiledTypes, expression.ToPBFieldType(retType))
```

Comment:
may I ask why we have both `probeFiledTypes` and `buildFiledTypes` if they are same?

Reply:
They are copied from `HashJoinExec`, actually in TiFlash only one is needed, I set both `probeFieldTypes` and `buildFieldTypes` in `DAGRequest` just to keep it the same with TiDB.